include/argparse/argparse.hpp

```diff
@@ -327,6 +328,43 @@ template <class T> struct parse_number<T, chars_format::fixed> {
 
 } // namespace details
 
+class SizeRange {
+  std::size_t m_min;
+  std::size_t m_max;
+
+public:
+  SizeRange(std::size_t minimum, std::size_t maximum) : m_min(minimum), m_max(maximum) {
+    if (minimum > maximum)
+      throw std::logic_error("Range of number of arguments is invalid");
+  }
+
+  bool contains(std::size_t value) const {
+    return value >= m_min && value <= m_max;
+  }
+
+  bool is_exact() const {
+    return m_min == m_max;
+  }
+
+  bool is_right_bounded() const {
+    return m_max < std::numeric_limits<std::size_t>::max();
+  }
+
+  std::size_t get_min() const {
+    return m_min;
+  }
+
+  std::size_t get_max() const {
+    return m_max;
+  }
+};
+
+enum class NArgsPattern {
+  ZeroOrOne,
```

Comment:
For consistency, let's use lower_case underscore-separated field names for `enum class`es.

Reply:
Fixed in https://github.com/p-ranav/argparse/pull/125/commits/b869b5a209f6f2963988fbcd5a7ec62ddd60d573